modules/build/src/main/scala/scala/build/input/Inputs.scala

```diff
@@ -263,14 +263,22 @@ object Inputs {
     stdinOpt: => Option[Array[Byte]],
     acceptFds: Boolean,
     enableMarkdown: Boolean,
-    isRunWithShebang: Boolean
+    programInvokeData: ScalaCliInvokeData
   ): Seq[Either[String, Seq[Element]]] = args.zipWithIndex.map {
     case (arg, idx) =>
       lazy val path      = os.Path(arg, cwd)
       lazy val dir       = path / os.up
       lazy val subPath   = path.subRelativeTo(dir)
       lazy val stdinOpt0 = stdinOpt
       lazy val content   = os.read.bytes(path)
+      val (isRunWithShebang, isRunWithDefault) = programInvokeData.subCommand match
+        case SubCommand.Shebang => (true, false)
+        case SubCommand.Default => (false, true)
+        case _                  => (false, false)
+      lazy val fullProgramCall = programInvokeData.progName +
+        s"${if isRunWithDefault then "" else s" ${programInvokeData.subCommandName}"}"
+      lazy val progName = os.Path(programInvokeData.progName, cwd).baseName
```

Comment:
What is the value of `progName` if someone is running `scala-cli` from PATH? For example:
```
scala-cli shebang script
```

Reply:
It's `scala-cli` when running from PATH. But actually I think it would be better to use `baseRunnerName` instead of `progName`, as `progName` may be a path so the output of errors will be less readable. What do you think?

Reply:
Will be sufficient if we use `progName`.